src/buffer.rs

```diff
@@ -344,18 +344,18 @@ pub struct CodedBufferSegment<'s> {
 
 /// Helper to access segments of mapped coded buffer
 pub struct CodedBuffer<'p, 's: 'p, D: SurfaceMemoryDescriptor, T: Borrow<Surface<D>>> {
-    buffer: Buffer,
     segments: Vec<CodedBufferSegment<'s>>,
-    _picture: &'p Picture<PictureSync, T>,
+    picture: &'p Picture<PictureSync, T>,
     _phantom_data: std::marker::PhantomData<D>,
 }
 
 impl<'p, 's: 'p, D: SurfaceMemoryDescriptor, T: Borrow<Surface<D>>> CodedBuffer<'p, 's, D, T> {
     /// Map a 'VAEncCodedBufferType' buffer. `picture` makes sure that the surface was synced and
     /// would not change state until this buffer is unmapped (dropped).
-    pub fn new(buffer: Buffer, picture: &'p Picture<PictureSync, T>) -> Result<Self, VaError> {
+    pub fn new(picture: &'p Picture<PictureSync, T>) -> Result<Self, VaError> {
         let mut addr = std::ptr::null_mut();
         let mut segments = Vec::new();
+        let buffer = picture.coded_buffer().unwrap();
```

Comment:
Let's return a relevant VaError instead of crashing here.

Reply:
Done in a separate commit. PTAL :smile: